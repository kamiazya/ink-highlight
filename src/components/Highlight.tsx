import React, { FC, useContext } from 'react';
import PropTypes from 'prop-types';
import { highlight } from 'cli-highlight';
import { HighlightOptions, HighlightOptionsContext } from './HighlightOptionsContext';
import { HighlightThemeContext } from './HighlightThemeContext';

export type Props = {
  code: string;
} & HighlightOptions;

export const Highlight: FC<Props> = ({ code, ...option }) => {
  const theme = useContext(HighlightThemeContext);
  const contextOption = useContext(HighlightOptionsContext);
  return (
    <>
      {highlight(code, {
        ...contextOption,
        ...option,
        theme,
      })}
    </>
  );
};

Highlight.propTypes = {
  code: PropTypes.string.isRequired,
  language: PropTypes.string,
  ignoreIllegals: PropTypes.bool,
  continuation: PropTypes.any,
  languageSubset: PropTypes.arrayOf(PropTypes.string.isRequired),
};

Highlight.defaultProps = {
  language: undefined,
  ignoreIllegals: undefined,
  continuation: undefined,
  languageSubset: undefined,
};
