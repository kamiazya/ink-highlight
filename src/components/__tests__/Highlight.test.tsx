import React from 'react';
import { render } from 'ink-testing-library';
import { wrap } from 'jest-snapshot-serializer-raw';
import { Highlight } from '../Highlight';

describe('Highlight Component', () => {
  test.each([
    ['SELECT * FROM users;', 'sql'],
    ["console.log('Hello world!!')", 'js'],
    ["console.log('Hello world!!')", undefined],
  ])('How cli-highlight is called', (code, language) => {
    const { lastFrame } = render(<Highlight code={code} language={language} />);
    expect(wrap(lastFrame())).toMatchSnapshot();
  });

  test('Change code and language', () => {
    const { lastFrame, rerender } = render(<Highlight code="SELECT * FROM users;" language="sql" />);
    expect(wrap(lastFrame())).toMatchSnapshot();

    rerender(<Highlight code="console.log('Hello world!!')" language="js" />);
    expect(wrap(lastFrame())).toMatchSnapshot();
  });
});
