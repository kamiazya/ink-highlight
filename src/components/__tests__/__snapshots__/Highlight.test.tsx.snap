// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight Component Change code and language 1`] = `highlight("SELECT * FROM users;", {"language":"sql"})`;

exports[`Highlight Component Change code and language 2`] = `highlight("console.log('Hello world!!')", {"language":"js"})`;

exports[`Highlight Component How cli-highlight is called 1`] = `highlight("SELECT * FROM users;", {"language":"sql"})`;

exports[`Highlight Component How cli-highlight is called 2`] = `highlight("console.log('Hello world!!')", {"language":"js"})`;

exports[`Highlight Component How cli-highlight is called 3`] = `highlight("console.log('Hello world!!')", {})`;
